feat: add FeeReport types to type definitions

yahgw-ai opened the pull request:
## Summary
- Added FeeReport interface to type definitions as specified in issue #190
- Implemented comprehensive test coverage for the new types
- Followed TDD methodology throughout implementation

## Changes
- Added `FeeReport` interface to `src/types/index.ts` with:
  - `metadata` property containing `chain_id`
  - `distributors` mapping with distributor address as key
  - Each distributor has `name` and `daily_totals` array
  - Daily totals include all required fields: `date`, `start_balance_wei`, `end_balance_wei`, `balance_change_wei`, `distributions_wei`, `distributions_count`, `total_wei`
  - All wei values use string type for precision as required
- Added comprehensive tests to `__tests__/units/types.test.ts` covering:
  - Basic FeeReport structure
  - Multiple distributors with multiple daily totals
  - All required fields in daily_totals entries
  - String type validation for all wei values

## Test Plan
✅ All existing tests pass
✅ New tests added for FeeReport type
✅ TypeScript compilation successful
✅ ESLint and formatting checks pass

Closes #190

yahgw-ai committed:
feat: add FeeReport types to type definitions

Add FeeReport interface with metadata and distributors properties
Add daily_totals array structure with all required fields
Ensure all wei values use string type for precision
Add comprehensive tests for FeeReport type structure

Closes #190

yahgwai commented on src/types/index.ts:
```diff
@@ -87,6 +87,26 @@ export interface RecipientRecievedEventData {
+  };
+  distributors: {
+    [distributorAddress: string]: {
+      name: string; // Distributor name
```
The distributor doesnt need a name - lets get rid of this property. We can then also get rid of the daily_totals property and make object:
[distributorAddress: string] : Array<data>

yahgw-ai committed:
refactor: simplify FeeReport type structure per review feedback

Remove name property from distributors
Remove daily_totals wrapper - distributor address now maps directly to array
Update all tests to match new simpler structure

yahgw-ai committed:
docs: update fee-calculator spec to match simplified FeeReport type

Update type definition and example output to remove name property and daily_totals wrapper
Align spec with the actual implementation from PR feedback
